protecode/util.py

```diff
@@ -114,13 +116,25 @@ def group_resources(components):
             for resource_name, resources in resource_groups.items():
                 yield resources
 
+        def _filter_resources_to_scan(component, resource):
+            # check whether the trait was configured to filter out the resource
+            configured_image_reference_filter_response = image_reference_filter(component, resource)
+            if not configured_image_reference_filter_response:
+                return False
+
+            # check for scanning labels on resource in cd
+            if label := resource.find_label(name=sdo.labels.ScanLabelName.BINARY_SCAN):
```

Comment:
does this work? You are passing the enum value instead of a string. Do you handle the ValueError exception?